### ISSUE TITLE
Add `build` key support for v3 of Docker Compose

### ISSUE THREAD

**Issue by cdrage** [kubernetes/kompose#636]
This will have to be fixed upstream as the code here in `docker/cli` ignores build at the moment: https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9

This is a tracking issue as #600 has yet to be merged.

**Comment by surajssd** [comment]
@cdrage #600 is merged!

**Comment by cdrage** [comment]
@surajssd Yes, but this issue is for post-# #600 merge. V3 Build support does not work due to: https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9 (see ignored section)

**Comment by surajnarwade** [comment]
@surajssd @cdrage , as `build` is still unsupported in https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9 , shall we use it from v1v2.go for v3.go ?
does it make sense ? if yes I am up for this :)

**Comment by cdrage** [comment]
@surajnarwade Bit confused what you mean.

Yes, we are indeed going to be re-using the code from `v1v2.go` in `v3.go` but that's the easy part, the problem is parsing v3 correctly so we can actually retrieve the information.

This would have to be fixed upstream unfortunately, or we can find a work-around to retrieve the keys.

**Comment by cdrage** [comment]
When the time comes, we will move the `build` conversion code to a separate file (`compose.go` maybe where the other functions are? So both v1v2.go and v3.go can use it.

If you can find a way to retrieve the build key that'd be great! :+1: :100: :+1:

**Comment by surajnarwade** [comment]
thanks , @cdrage I will give it a try :)

**Comment by cdrage** [comment]
I think we should set this as high-priority so we can add `build` support for Compose V3 as that is the *only* thing that is missing between v1v2 and v3.. Unfortunately it's due to the key being missing  from https://github.com/docker/cli/blob/master/cli/compose/types/types.go#L9 so we can use it to parse Docker Compose.

**Comment by cdrage** [comment]
PR pushed here: https://github.com/docker/cli/pull/481

**Comment by coli** [comment]
Looks like it doesn't support just a string for build.

* 'Build' expected a map, got 'string'

**Comment by surajnarwade** [comment]
@coli , yeah :( 
need to fix that in docker/cli

**Comment by surajnarwade** [comment]
@coli , that problem will resolve after https://github.com/docker/libcompose/pull/498 gets merged and again vendoring for `libcompose` and `docker/cli`

**Comment by cdrage** [comment]
@surajnarwade No idea why you're referencing docker/libcompose. Docker Compose V3 support within Kompose is using docker/cli

Your PR is here: https://github.com/kubernetes/kompose/pull/846 for Version 3 implementation. 

@coli are you using Version 2 or Version 3 in your Docker Compose YAML file?

If you're using Version 2, this is a separate issue as Version 3 is yet to be supported.

**Comment by surajnarwade** [comment]
@cdrage , @coli  is getting issue for v3, to solve this, I need to update `Docker CLI` vendor and then `libcompose` to solve `logrus` issues

**Comment by coli** [comment]
I'm on v3.

**Comment by cdrage** [comment]
Okay, so your statement about libcompose was unrelated.. The problem will *not* be resolved after https://github.com/docker/libcompose/pull/498 is merged in. It will be resolved after #846

**Comment by surajnarwade** [comment]
@cdrage to merge #846 , I need to update vendor for `docker/cli` upon which kompose is breaking due to `logrus` stuff

**Comment by tsauvajon** [comment]
How do you fix `'Build' expected a map, got 'string'` ?

**Comment by rkachowski** [comment]
@tsauvajon I've had this same problem, it looks like kompose doesn't respect the optional string parameter. You can work around it by changing the config to a map like so
```yaml
  build:
    context: ./build_path
```

**Comment by tsauvajon** [comment]
@rkachowski oh okay I had not understood the error message thanks a lot!
